prototypes/technology.lua

```diff
@@ -116,5 +120,167 @@ data:extend(
+		unit =
+		{
+			count = 100,
+			ingredients =
+			{
+				{ "automation-science-pack", 1 },
+				{ "logistic-science-pack", 1 },
+			},
+			time = 30
+		},
+		order = "c-a"
+	},
+	-- Logistics Warehouse Research 1
+	{
+		type = "technology",
+		name = "warehouse-logistics-research-1-tiny",
+		icon = "__Warehousing__/graphics/research/warehouse-logistics-research-tiny.png",
+		icon_size = 128,
+		effects =
+		{
+			{
+				type = "unlock-recipe",
+				recipe = "warehouse-passive-provider-small",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "warehouse-storage-small",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "storehouse-passive-provider-small",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "storehouse-storage-small",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "warehouse-passive-provider-tiny",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "warehouse-storage-tiny",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "storehouse-passive-provider-tiny",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "storehouse-storage-tiny",
+			},
+		},
+		prerequisites = { "warehouse-research-tiny", "robotics" },
+		unit =
+		{
+			count = 300,
+			ingredients = {
+				{ "automation-science-pack", 1 },
+				{ "logistic-science-pack", 1 },
+			},
+			time = 30
+		},
+		order = "c-k-a"
+	},
+	-- Logistics Warehouse Research 2
+	{
+		type = "technology",
+		name = "warehouse-logistics-research-2-tiny",
+		icon = "__Warehousing__/graphics/research/warehouse-logistics-research-tiny.png",
+		icon_size = 128,
+		effects =
+		{
+			{
+				type = "unlock-recipe",
+				recipe = "warehouse-active-provider-small",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "warehouse-requester-small",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "warehouse-buffer-small",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "storehouse-active-provider-small",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "storehouse-requester-small",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "storehouse-buffer-small",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "warehouse-active-provider-tiny",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "warehouse-requester-tiny",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "warehouse-buffer-tiny",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "storehouse-active-provider-tiny",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "storehouse-requester-tiny",
+			},
+			{
+				type = "unlock-recipe",
+				recipe = "storehouse-buffer-tiny",
+			},
+		},
+		prerequisites = { "warehouse-logistics-research-1-tiny", "logistic-system" },
+		unit =
+		{
+			count = 600,
+			ingredients = {
+				{ "automation-science-pack", 1 },
+				{ "logistic-science-pack", 1 },
+				{ "chemical-science-pack", 1 },
+				{ "utility-science-pack", 1 },
+			},
+			time = 30
+		},
+		order = "c-k-b"
+	},
+})
```

Comment:
Newline at end of file, please.

Reply:
New Line like leaving a blank line? Is that a common unwritten rule for coding? If so that's totally fine I just didn't know haha told you I'm a noob :D

Reply:
Git complains when there is no newline sequence at the end of a file. It's a convention followed by many, but not all, popular text editors. I prefer to always have it, because otherwise Git commands can be noisier than necessary and distract from the actual changes.